src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -1558,11 +1558,23 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
         /// Checks to see if a body is legal given the current modifiers.
         /// If it is not, a diagnostic is added with the current type.
         /// </summary>
-        protected void CheckModifiersForBody(Location location, DiagnosticBag diagnostics)
+        protected void CheckModifiersForBody(SyntaxNode syntax, Location location, DiagnosticBag diagnostics)
```

Comment:
>syntax [](start = 56, length = 6)

Nit: could you leave a comment or maybe rename the parameters to clarify what is `syntax` pointing out versus `location`?